### ISSUE TITLE
Plugins: Unable to load signed datasource plugin when upgrading it from unsigned version

### ISSUE THREAD

**Issue by linoecarrillo** [grafana/grafana#29950]
**What happened**:
Upgraded datasource plugin from version that was unsigned to one that was signed, but Grafana still saw it as unsigned.
```
installing sbueringer-consul-datasource @ 0.2.1
--
from: https://grafana.com/api/plugins/sbueringer-consul-datasource/versions/0.2.1/download
into: /var/lib/grafana/plugins
✔ Installed sbueringer-consul-datasource successfully
Restart grafana after installing plugins . <service grafana-server restart>
--
t=2020-12-16T01:30:37+0000 lvl=info msg="Starting Grafana" logger=server version=7.3.5 commit=11f305f88a branch=HEAD compiled=2020-12-09T16:02:57+0000
...
t=2020-12-16T01:30:37+0000 lvl=warn msg="Some plugins failed to load" logger=plugins errors="[plugin \"sbueringer-consul-datasource\" is unsigned]"
```
OR alternatively:

Install 0.1.5 - `bin/grafana-cli plugins install sbueringer-consul-datasource`
Then install 0.2.1 - `bin/grafana-cli pl--pluginUrl='https://grafana.com/api/plugins/sbueringer-consul-datasource/versions/0.2.1/download'  plugins install sbueringer-consul-datasource/`

**What you expected to happen**:
Expected Grafana to realize that the new version of the datasource plugin was signed and then load it without issues.

**How to reproduce it (as minimally and precisely as possible)**:
1. `docker run --rm -it -p 3000:3000 --name=grafana -e "GF_INSTALL_PLUGINS=sbueringer-consul-datasource 0.1.5" -v /some/host/path:/var/lib/grafana grafana/grafana:7.3.5`
1. Ctrl+C out of the container
1. `docker run --rm -it -p 3000:3000 --name=grafana -e "GF_INSTALL_PLUGINS=sbueringer-consul-datasource 0.2.1" -v /some/host/path:/var/lib/grafana grafana/grafana:7.3.5`
1. Afterwards, I remove the `plugins/sbueringer-consul-datasource` directory inside the bind-mounted volume, run step 3 again and it is then able load the 0.2.1 plugin.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.3.5
- Data source type & version: sbueringer-consul-datasource 0.1.5 (unsigned), 0.2.1 (signed)
- OS Grafana is installed on: official Grafana Docker image
- User OS & Browser: N/A
- Grafana plugins: N/A
- Others: N/A

**Comment by wbrowne** [comment]
Thanks for creating this issue @linoecarrillo - looks like a bug alright. Will discuss with my colleagues to see if this is a known design flaw and figure out the next steps. Thanks again!

**Comment by marefr** [comment]
@wbrowne seems like this should have been closed by #31515. Is that correct?

**Comment by wbrowne** [comment]
Yep exactly @marefr. This fix should be rolled out with Grafana 8.0, so will close this issue
